inference-engine/src/readers/reader_api/ie_reader.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <details/ie_irelease.hpp>
+#include <cpp/ie_cnn_network.h>
+#include <ie_iextension.h>
+#include <istream>
+#include <string>
+#include <vector>
+#include <ie_api.h>
+
+namespace InferenceEngine {
+/**
+ * @def INFERENCE_ENGINE_READER_API(TYPE)
+ * @brief Defines Inference Engine Extension API method
+ */
+
+#if defined(_WIN32) && defined(IMPLEMENT_INFERENCE_ENGINE_READER_API)
+#define INFERENCE_ENGINE_READER_API(TYPE) extern "C" __declspec(dllexport) TYPE
+#else
+#define INFERENCE_ENGINE_READER_API(TYPE) INFERENCE_ENGINE_API(TYPE)
+#endif
+
+
+/**
+ * @brief IReader an abstract interface for Inference Engine readers
+ */
+class INFERENCE_ENGINE_API_CLASS(IReader): public details::IRelease {
+public:
+    /**
+     * @brief Checks that reader supports format of the model
+     * @param model stream with model
+     * @return true if format is supported
+     */
+    virtual bool supportModel(std::istream& model) const = 0;
+    /**
+     * @brief Reads the model to CNNNetwork
+     * @param model stream with model
+     * @param exts vector with extensions
+     *
+     * @return CNNNetwork
+     */
+    virtual CNNNetwork read(std::istream& model, const std::vector<IExtensionPtr>& exts) const = 0;
+    /**
+     * @brief Reads the model to CNNNetwork
+     * @param model stream with model
+     * @param weights stream with binary data
+     * @param exts vector with extensions
+     *
+     * @return CNNNetwork
+     */
+    virtual CNNNetwork read(std::istream& model, std::istream& weights, const std::vector<IExtensionPtr>& exts) const = 0;
+
+    /**
+     * @brief Returns all supported extensions for data files
+     *
+     * @return vector of file extensions
+     */
+    virtual std::vector<std::string> getDataFileExts() const = 0;
```

Comment:
Can we provide examples if docs? What are the data extensions depending on different frameworks

Reply:
Added an example